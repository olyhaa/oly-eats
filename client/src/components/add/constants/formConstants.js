const ID = 'id';
const TITLE = 'title';
const DESCRIPTION = 'description';
const PHOTO_URL = 'photo';
const SERVINGS = 'servings';
const DATE_ADDED = 'dateAdded';
const LAST_UPDATED = 'lastUpdated';
const TAGS = 'tags';

// INGREDIENTS
const INGREDIENTS = 'ingredients';
const INGREDIENTS_LABEL = 'ingredientsLabel';
const INGREDIENTS_LIST = 'ingredientsList';

// DIRECTIONS
const DIRECTIONS = 'directions';
const DIRECTIONS_LABEL = 'directionsLabel';
const DIRECTIONS_LIST = 'directionsList';

// SOURCE
const SOURCE_DISPLAY = 'sourceDisplay';
const SOURCE_URL = 'sourceURL';

// TIMING
const TIMING_PREP_VALUE_HOURS = 'timingPrepValueHours';
const TIMING_PREP_VALUE_MINS = 'timingPrepValueMins';
const TIMING_TOTAL_VALUE_HOURS = 'timingTotalValueHours';
const TIMING_TOTAL_VALUE_MINS = 'timingTotalValueMins';

export const FIELDS = {
  ID,
  TITLE,
  DESCRIPTION,
  PHOTO_URL,
  SOURCE_DISPLAY,
  SOURCE_URL,
  SERVINGS,
  TIMING_PREP_VALUE_HOURS,
  TIMING_PREP_VALUE_MINS,
  TIMING_TOTAL_VALUE_HOURS,
  TIMING_TOTAL_VALUE_MINS,
  INGREDIENTS,
  INGREDIENTS_LABEL,
  INGREDIENTS_LIST,
  DIRECTIONS,
  DIRECTIONS_LABEL,
  DIRECTIONS_LIST,
  TAGS,
  DATE_ADDED,
  LAST_UPDATED,
};

export const requiredFields = [
  FIELDS.TITLE,
  FIELDS.DESCRIPTION,
  FIELDS.SOURCE_DISPLAY,
  FIELDS.SERVINGS,
];

export const isRequired = (field) => {
  return requiredFields.includes(field);
};

export const isField = (field) => {
  const fieldKeys = Object.keys(FIELDS);
  for (let i = 0; i < fieldKeys.length; i++) {
    if (FIELDS[fieldKeys[i]] === field) {
      return true;
    }
  }
  return false;
};
