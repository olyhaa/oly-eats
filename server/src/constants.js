const CATEGORY = 'category';
const CUISINE = 'cuisine';
const EQUIPMENT = 'equipment';
const MEAL_TYPE = 'meal_type';

export const TAGS = {
  CATEGORY,
  CUISINE,
  EQUIPMENT,
  MEAL_TYPE,
};
