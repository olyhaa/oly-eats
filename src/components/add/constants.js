const TITLE = 'title';
const DESCRIPTION = 'description';
const PHOTO_URL = 'photo';
const SERVINGS = 'servings';

// INGREDIENTS
const INGREDIENTS = 'ingredients';
const INGREDIENTS_LABEL = 'ingredientsLabel';
const INGREDIENTS_LIST = 'ingredientsList';

// DIRECTIONS
const DIRECTIONS = 'directions';
const DIRECTIONS_LABEL = 'directionsLabel';
const DIRECTIONS_LIST = 'directionsList';

// SOURCE
const SOURCE_DISPLAY = 'sourceDisplay';
const SOURCE_URL = 'sourceURL';

// TIMING
const TIMING_PREP_VALUE_HOURS = 'timingPrepValueHours';
const TIMING_PREP_VALUE_MINS = 'timingPrepValueMins';
const TIMING_TOTAL_VALUE_HOURS = 'timingTotalValueHours';
const TIMING_TOTAL_VALUE_MINS = 'timingTotalValueMins';

// OPTIONAL FIELDS
const MEAL_TYPE = 'mealType';
const CUISINE = 'cuisine';
const EQUIPMENT = 'equipment';
const CATEGORY = 'category';

export const FIELDS = {
  TITLE,
  DESCRIPTION,
  PHOTO_URL,
  SOURCE_DISPLAY,
  SOURCE_URL,
  SERVINGS,
  TIMING_PREP_VALUE_HOURS,
  TIMING_PREP_VALUE_MINS,
  TIMING_TOTAL_VALUE_HOURS,
  TIMING_TOTAL_VALUE_MINS,
  INGREDIENTS,
  INGREDIENTS_LABEL,
  INGREDIENTS_LIST,
  DIRECTIONS,
  DIRECTIONS_LABEL,
  DIRECTIONS_LIST,
  MEAL_TYPE,
  CUISINE,
  EQUIPMENT,
  CATEGORY,
};

export const requiredFields = [
  FIELDS.TITLE,
  FIELDS.DESCRIPTION,
  FIELDS.SOURCE_DISPLAY,
  FIELDS.SERVINGS,
  FIELDS.INGREDIENTS_LIST,
  FIELDS.DIRECTIONS_LIST,
];

export const isRequired = (field) => {
  return requiredFields.includes(field);
};

export const TIMING_UNITS = {
  MINUTES: 'minutes',
  HOURS: 'hours',
};

export const isField = (field) => {
  const fieldKeys = Object.keys(FIELDS);
  for (let i = 0; i < fieldKeys.length; i++) {
    if (FIELDS[fieldKeys[i]] === field) {
      return true;
    }
  }
  return false;
};
