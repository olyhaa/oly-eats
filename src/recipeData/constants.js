const TEASPOON = 'teaspoon';
const TEASPOON_SHORT = 't';
const TABLESPOON = 'tablespoon';
const TABLESPOON_SHORT = 'T';
const CUP = 'cup';
const CUP_SHORT = 'c';
const OUNCE = 'ounce';
const OUNCE_SHORT = 'oz';
const POUND = 'pound';
const POUND_SHORT = 'lb';

const UNITS_EXACT = [
  TEASPOON_SHORT,
  TABLESPOON_SHORT,
  CUP_SHORT,
  OUNCE_SHORT,
  POUND_SHORT
];

const UNITS_MATCH = [TEASPOON, TABLESPOON, CUP, OUNCE, POUND];

export const isUnit = value => {
  let matches;
  // check if it's shorthand (must match exactly)
  matches = UNITS_EXACT.filter(unit => {
    return unit === value;
  });
  if (matches.length > 0) {
    return true;
  }

  // check if it's longhand
  matches = UNITS_MATCH.filter(unit => {
    return value.toUpperCase().includes(unit.toUpperCase());
  });
  if (matches.length > 0) {
    return true;
  }
  return false;
};
