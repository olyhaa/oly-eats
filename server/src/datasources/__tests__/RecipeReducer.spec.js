import {
  timingReducer,
  timeReducer,
  directionsReducer,
  directionStepsReducer,
  ingredientsReducer,
  ingredientReducer,
  rangedAmountReducer,
  tagsReducer,
  recipeReducer,
  recipeMutationReducer,
  metaReducer,
} from '../RecipeReducer';
import { TIMINGS, TIME_UNITS } from '../../constants';

const mockTimings = [
  {
    id: 1,
    recipeId: 123,
    value: '2',
    units: TIME_UNITS.MINUTE,
    type: TIMINGS.PREP_TIME,
    createdAt: '2020-05-10 00:02:45.511 +00:00',
    updatedAt: '2020-05-10 00:03:45.511 +00:00',
  },
  {
    id: 2,
    recipeId: 123,
    value: '20',
    units: TIME_UNITS.MINUTE,
    type: TIMINGS.TOTAL_TIME,
    createdAt: '2020-05-10 00:04:45.511 +00:00',
    updatedAt: '2020-05-10 00:05:45.511 +00:00',
  },
  {
    id: 3,
    recipeId: 123,
    value: '1',
    units: TIME_UNITS.HOUR,
    type: TIMINGS.TOTAL_TIME,
    createdAt: '2020-05-10 00:06:45.511 +00:00',
    updatedAt: '2020-05-10 00:07:45.511 +00:00',
  },
];

const mockDirectionStep = [
  {
    id: 1,
    text: 'step 1',
    directionSectionId: 2,
    createdAt: '2020-05-10 00:09:45.511 +00:00',
    updatedAt: '2020-05-10 00:10:45.511 +00:00',
  },
  {
    id: 2,
    text: 'step 2',
    directionSectionId: 2,
    createdAt: '2020-05-10 00:11:45.511 +00:00',
    updatedAt: '2020-05-10 00:12:45.511 +00:00',
  },
  {
    id: 3,
    text: 'step 1',
    directionSectionId: 1,
    createdAt: '2020-05-10 00:09:45.511 +00:00',
    updatedAt: '2020-05-10 00:10:45.511 +00:00',
  },
];

const fullRecipe = {
  recipe: {
    id: 123,
    title: 'my first recipe',
    description: 'the best recipe ever',
    source_display: "Mandy's Kitchen",
    source_url: 'http://some.url.com/recipe',
    photo_url: 'http://some.url.com/pic.jpg',
    servings: 2,
    createdAt: '2020-05-10 00:00:45.511 +00:00',
    updatedAt: '2020-05-10 00:01:45.511 +00:00',
    timings: [
      {
        id: 1,
        recipeId: 123,
        value: '2',
        units: 'MINUTE',
        type: 'PREP',
        createdAt: '2020-05-10 00:02:45.511 +00:00',
        updatedAt: '2020-05-10 00:03:45.511 +00:00',
      },
      {
        id: 2,
        recipeId: 123,
        value: '20',
        units: 'MINUTE',
        type: 'TOTAL',
        createdAt: '2020-05-10 00:04:45.511 +00:00',
        updatedAt: '2020-05-10 00:05:45.511 +00:00',
      },
      {
        id: 3,
        recipeId: 123,
        value: '1',
        units: 'HOUR',
        type: 'TOTAL',
        createdAt: '2020-05-10 00:06:45.511 +00:00',
        updatedAt: '2020-05-10 00:07:45.511 +00:00',
      },
    ],
    directionSections: [
      {
        id: 1,
        label: 'section 1',
        recipeId: 123,
        createdAt: '2020-05-10 00:07:45.511 +00:00',
        updatedAt: '2020-05-10 00:08:45.511 +00:00',
        directionSteps: [
          {
            id: 1,
            text: 'step 1',
            directionSectionId: 2,
            createdAt: '2020-05-10 00:09:45.511 +00:00',
            updatedAt: '2020-05-10 00:10:45.511 +00:00',
          },
          {
            id: 2,
            text: 'step 2',
            directionSectionId: 2,
            createdAt: '2020-05-10 00:11:45.511 +00:00',
            updatedAt: '2020-05-10 00:12:45.511 +00:00',
          },
        ],
      },
    ],
    ingredientSections: [
      {
        id: 1,
        label: 'ingredient section 1',
        recipeId: 102,
        createdAt: '2020-05-10 00:07:45.511 +00:00',
        updatedAt: '2020-05-10 00:08:45.511 +00:00',
        ingredients: [
          {
            id: 1,
            amount: '2',
            unit: 'cup',
            prep: 'chopped',
            name: 'apples',
            toTaste: true,
            optional: true,
            ingredientSectionId: 2,
            createdAt: '2020-05-10 00:07:45.511 +00:00',
            updatedAt: '2020-05-10 00:08:45.511 +00:00',
          },
          {
            id: 2,
            unit: 'tablespoon',
            prep: 'minced',
            name: 'garlic',
            toTaste: true,
            optional: false,
            ingredientSectionId: 2,
            createdAt: '2020-05-10 00:07:45.511 +00:00',
            updatedAt: '2020-05-10 00:08:45.511 +00:00',
            rangedAmount: {
              id: 1,
              min: '5',
              max: '10',
              createdAt: '2020-05-10 00:07:45.511 +00:00',
              updatedAt: '2020-05-10 00:08:45.511 +00:00',
              ingredientId: 2,
            },
          },
        ],
      },
    ],
    tags: [
      {
        id: 1,
        label: 'bread',
        reatedAt: '2020-05-22T22:25:23.214Z',
        updatedAt: '2020-05-22T22:25:23.214Z',
        tagTypeId: 2,
        recipeTag: {
          createdAt: '2020-05-23T00:20:55.988Z',
          updatedAt: '2020-05-23T00:20:55.988Z',
          recipeId: 123,
          tagId: 1,
        },
        tagType: {
          id: 2,
          label: 'category',
          createdAt: '2020-05-22T22:25:14.275Z',
          updatedAt: '2020-05-22T22:25:14.275Z',
        },
      },
      {
        id: 2,
        label: 'snack',
        reatedAt: '2020-05-22T22:25:23.214Z',
        updatedAt: '2020-05-22T22:25:23.214Z',
        tagTypeId: 4,
        recipeTag: {
          createdAt: '2020-05-23T00:20:55.988Z',
          updatedAt: '2020-05-23T00:20:55.988Z',
          recipeId: 4,
          tagId: 2,
        },
        tagType: {
          id: 4,
          label: 'meal_type',
          createdAt: '2020-05-22T22:25:14.275Z',
          updatedAt: '2020-05-22T22:25:14.275Z',
        },
      },
    ],
    isFavorite: true,
  },
};

describe('recipeReducer', () => {
  it('no data', () => {
    expect(recipeReducer({})).toBeNull();
  });

  it('missing recipe component', () => {
    const recipe = Object.assign({}, fullRecipe);
    recipe.recipe = undefined;
    expect(recipeReducer({ ...recipe })).toBeNull();
  });

  it('all recipe components', () => {
    expect(recipeReducer(fullRecipe)).toMatchSnapshot();
  });
});

describe('recipeMutationReducer', () => {
  it('no success field', () => {
    expect(
      recipeMutationReducer({ message: 'Success!', recipe: fullRecipe })
    ).toMatchSnapshot();
  });

  it('no message field', () => {
    expect(
      recipeMutationReducer({ success: true, recipe: fullRecipe })
    ).toMatchSnapshot();
  });

  it('no recipe field', () => {
    expect(
      recipeMutationReducer({
        success: true,
        message: 'Success!',
      })
    ).toMatchSnapshot();
  });

  it('all fields', () => {
    expect(
      recipeMutationReducer({
        success: true,
        message: 'Success!',
        recipe: fullRecipe,
      })
    ).toMatchSnapshot();
  });
});

describe('timeReducer', () => {
  it('properly reduces total time array', () => {
    const timeArray = [mockTimings[1], mockTimings[2]];
    expect(timeReducer({ timeArray })).toMatchSnapshot();
  });
});

describe('timingReducer', () => {
  it('properly reduces timing array', () => {
    expect(timingReducer({ timingArray: mockTimings })).toMatchSnapshot();
  });
});

describe('directionsReducer', () => {
  it('properly reduces direction array', () => {
    const directionSections = [
      {
        createdAt: '2020-05-10 00:07:45.511 +00:00',
        id: '1',
        label: 'section 1',
        recipeId: '123',
        directionSteps: [
          {
            createdAt: '2020-05-10 00:09:45.511 +00:00',
            id: '3',
            directionSectionId: '1',
            text: 'step 1',
            updatedAt: '2020-05-10 00:10:45.511 +00:00',
          },
        ],
        updatedAt: '2020-05-10 00:08:45.511 +00:00',
      },
      {
        createdAt: '2020-05-10 00:07:45.511 +00:00',
        id: '2',
        recipeId: '123',
        directionSteps: [
          {
            createdAt: '2020-05-10 00:09:45.511 +00:00',
            id: '1',
            directionSectionId: '2',
            text: 'step 1',
            updatedAt: '2020-05-10 00:10:45.511 +00:00',
          },
          {
            createdAt: '2020-05-10 00:11:45.511 +00:00',
            id: '2',
            directionSectionId: '2',
            text: 'step 2',
            updatedAt: '2020-05-10 00:12:45.511 +00:00',
          },
        ],
        updatedAt: '2020-05-10 00:08:45.511 +00:00',
      },
    ];
    expect(directionsReducer({ directionSections })).toMatchSnapshot();
  });
});

describe('directionStepsReducer', () => {
  it('properly reduces direction step array', () => {
    expect(
      directionStepsReducer({ steps: mockDirectionStep })
    ).toMatchSnapshot();
  });
});

describe('ingredientsReducer', () => {
  it('properly reduces ingredients array', () => {
    const ingredientSections = [
      {
        createdAt: '2020-05-10 00:07:45.511 +00:00',
        id: '1',
        label: 'section 1',
        recipeId: '123',
        ingredients: [
          {
            createdAt: '2020-05-10 00:09:45.511 +00:00',
            id: '3',
            ingredientSectionId: '1',
            amount: '2',
            unit: 'cup',
            prep: 'chopped',
            name: 'apples',
            toTaste: true,
            optional: true,
            updatedAt: '2020-05-10 00:10:45.511 +00:00',
          },
        ],
        updatedAt: '2020-05-10 00:08:45.511 +00:00',
      },
      {
        createdAt: '2020-05-10 00:07:45.511 +00:00',
        id: '2',
        recipeId: '123',
        ingredients: [
          {
            createdAt: '2020-05-10 00:09:45.511 +00:00',
            id: '1',
            ingredientSectionId: '2',
            unit: 'tablespoon',
            prep: 'minced',
            name: 'garlic',
            toTaste: true,
            optional: false,
            rangedAmount: {
              id: '1',
              min: '5',
              max: '10',
              createdAt: '2020-05-10 00:07:45.511 +00:00',
              updatedAt: '2020-05-10 00:08:45.511 +00:00',
              ingredientId: '1',
            },
            updatedAt: '2020-05-10 00:10:45.511 +00:00',
          },
          {
            createdAt: '2020-05-10 00:11:45.511 +00:00',
            id: '2',
            ingredientSectionId: '2',
            amount: '3',
            name: 'carrot',
            updatedAt: '2020-05-10 00:12:45.511 +00:00',
          },
        ],
        updatedAt: '2020-05-10 00:08:45.511 +00:00',
      },
    ];
    expect(ingredientsReducer({ ingredientSections })).toMatchSnapshot();
  });
});

describe('ingredientReducer', () => {
  it('properly reduces ingredient', () => {
    const ingredient = {
      createdAt: '2020-05-10 00:09:45.511 +00:00',
      id: '1',
      ingredientSectionId: '2',
      unit: 'tablespoon',
      prep: 'minced',
      name: 'garlic',
      toTaste: true,
      optional: false,
      rangedAmount: {
        id: '1',
        min: '5',
        max: '10',
        createdAt: '2020-05-10 00:07:45.511 +00:00',
        updatedAt: '2020-05-10 00:08:45.511 +00:00',
        ingredientId: '1',
      },
      updatedAt: '2020-05-10 00:10:45.511 +00:00',
    };
    expect(ingredientReducer({ ingredient })).toMatchSnapshot();
  });
});

describe('rangedAmountReducer', () => {
  it('properly reduces ranged amount', () => {
    const rangedAmount = {
      id: '1',
      min: '5',
      max: '10',
      createdAt: '2020-05-10 00:07:45.511 +00:00',
      updatedAt: '2020-05-10 00:08:45.511 +00:00',
      ingredientId: '1',
    };
    expect(rangedAmountReducer({ rangedAmount })).toMatchSnapshot();
  });
});

describe('tagsReducer', () => {
  it('empty array', () => {
    const recipeTags = [];
    expect(tagsReducer({ recipeTags })).toEqual([]);
  });

  it('properly reduces tags', () => {
    const recipeTags = [
      {
        id: 1,
        label: 'bread',
        reatedAt: '2020-05-22T22:25:23.214Z',
        updatedAt: '2020-05-22T22:25:23.214Z',
        tagTypeId: 2,
        recipeTag: {
          createdAt: '2020-05-23T00:20:55.988Z',
          updatedAt: '2020-05-23T00:20:55.988Z',
          recipeId: 123,
          tagId: 1,
        },
        tagType: {
          id: 2,
          label: 'category',
          createdAt: '2020-05-22T22:25:14.275Z',
          updatedAt: '2020-05-22T22:25:14.275Z',
        },
      },
      {
        id: 2,
        label: 'snack',
        reatedAt: '2020-05-22T22:25:23.214Z',
        updatedAt: '2020-05-22T22:25:23.214Z',
        tagTypeId: 4,
        recipeTag: {
          createdAt: '2020-05-23T00:20:55.988Z',
          updatedAt: '2020-05-23T00:20:55.988Z',
          recipeId: 4,
          tagId: 2,
        },
        tagType: {
          id: 4,
          label: 'meal_type',
          createdAt: '2020-05-22T22:25:14.275Z',
          updatedAt: '2020-05-22T22:25:14.275Z',
        },
      },
    ];
    expect(tagsReducer({ recipeTags })).toMatchSnapshot();
  });
});

describe('metaReducer', () => {
  it('empty object', () => {
    const metaObj = {};
    expect(metaReducer(metaObj)).toEqual({});
  });

  it('different created and updated dates', () => {
    const metaObj = {
      createdAt: '2020-05-23 14:34:55.641 +00:00',
      updatedAt: '2020-05-23 14:34:55.641 +00:00',
    };
    expect(metaReducer(metaObj)).toEqual({
      dateAdded: metaObj.createdAt,
    });
  });

  it('same created and updated dates', () => {
    const metaObj = {
      createdAt: '2020-05-20 14:34:55.641 +00:00',
      updatedAt: '2020-05-23 14:34:55.641 +00:00',
    };
    expect(metaReducer(metaObj)).toEqual({
      dateAdded: metaObj.createdAt,
      dateUpdated: metaObj.updatedAt,
    });
  });
});
