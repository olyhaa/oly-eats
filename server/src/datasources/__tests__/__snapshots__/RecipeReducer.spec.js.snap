// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directionStepsReducer properly reduces direction step array 1`] = `
Array [
  Object {
    "text": "step 1",
  },
  Object {
    "text": "step 2",
  },
  Object {
    "text": "step 1",
  },
]
`;

exports[`directionsReducer properly reduces direction array 1`] = `
Array [
  Object {
    "label": "section 1",
    "steps": Array [
      Object {
        "text": "step 1",
      },
    ],
  },
  Object {
    "steps": Array [
      Object {
        "text": "step 1",
      },
      Object {
        "text": "step 2",
      },
    ],
  },
]
`;

exports[`ingredientReducer properly reduces ingredient 1`] = `
Object {
  "amount": Object {
    "max": "10",
    "min": "5",
  },
  "name": "garlic",
  "optional": false,
  "prep": "minced",
  "toTaste": true,
  "unit": "tablespoon",
}
`;

exports[`ingredientsReducer properly reduces ingredients array 1`] = `
Array [
  Object {
    "ingredients": Array [
      Object {
        "amount": "2",
        "name": "apples",
        "optional": true,
        "prep": "chopped",
        "toTaste": true,
        "unit": "cup",
      },
    ],
    "label": "section 1",
  },
  Object {
    "ingredients": Array [
      Object {
        "amount": Object {
          "max": "10",
          "min": "5",
        },
        "name": "garlic",
        "optional": false,
        "prep": "minced",
        "toTaste": true,
        "unit": "tablespoon",
      },
      Object {
        "amount": "3",
        "name": "carrot",
        "optional": undefined,
        "prep": undefined,
        "toTaste": undefined,
        "unit": undefined,
      },
    ],
  },
]
`;

exports[`rangedAmountReducer properly reduces ranged amount 1`] = `
Object {
  "max": "10",
  "min": "5",
}
`;

exports[`recipeMutationReducer all fields 1`] = `
Object {
  "message": "Success!",
  "recipe": Object {
    "dateAdded": "2020-05-10 00:00:45.511 +00:00",
    "dateUpdated": "2020-05-10 00:01:45.511 +00:00",
    "description": "the best recipe ever",
    "directions": Array [
      Object {
        "label": "section 1",
        "steps": Array [
          Object {
            "text": "step 1",
          },
          Object {
            "text": "step 2",
          },
        ],
      },
    ],
    "id": 123,
    "ingredients": Array [
      Object {
        "ingredients": Array [
          Object {
            "amount": "2",
            "name": "apples",
            "optional": true,
            "prep": "chopped",
            "toTaste": true,
            "unit": "cup",
          },
          Object {
            "amount": Object {
              "max": "10",
              "min": "5",
            },
            "name": "garlic",
            "optional": false,
            "prep": "minced",
            "toTaste": true,
            "unit": "tablespoon",
          },
        ],
        "label": "ingredient section 1",
      },
    ],
    "photo": "http://some.url.com/pic.jpg",
    "servings": 2,
    "source": Object {
      "display": "Mandy's Kitchen",
      "url": "http://some.url.com/recipe",
    },
    "tags": Array [
      Object {
        "id": 1,
        "label": "bread",
        "typeid": 2,
      },
      Object {
        "id": 2,
        "label": "snack",
        "typeid": 4,
      },
    ],
    "timing": Object {
      "prep": Array [
        Object {
          "units": "MINUTE",
          "value": "2",
        },
      ],
      "total": Array [
        Object {
          "units": "MINUTE",
          "value": "20",
        },
        Object {
          "units": "HOUR",
          "value": "1",
        },
      ],
    },
    "title": "my first recipe",
  },
  "success": true,
}
`;

exports[`recipeMutationReducer no message field 1`] = `
Object {
  "message": undefined,
  "recipe": Object {
    "dateAdded": "2020-05-10 00:00:45.511 +00:00",
    "dateUpdated": "2020-05-10 00:01:45.511 +00:00",
    "description": "the best recipe ever",
    "directions": Array [
      Object {
        "label": "section 1",
        "steps": Array [
          Object {
            "text": "step 1",
          },
          Object {
            "text": "step 2",
          },
        ],
      },
    ],
    "id": 123,
    "ingredients": Array [
      Object {
        "ingredients": Array [
          Object {
            "amount": "2",
            "name": "apples",
            "optional": true,
            "prep": "chopped",
            "toTaste": true,
            "unit": "cup",
          },
          Object {
            "amount": Object {
              "max": "10",
              "min": "5",
            },
            "name": "garlic",
            "optional": false,
            "prep": "minced",
            "toTaste": true,
            "unit": "tablespoon",
          },
        ],
        "label": "ingredient section 1",
      },
    ],
    "photo": "http://some.url.com/pic.jpg",
    "servings": 2,
    "source": Object {
      "display": "Mandy's Kitchen",
      "url": "http://some.url.com/recipe",
    },
    "tags": Array [
      Object {
        "id": 1,
        "label": "bread",
        "typeid": 2,
      },
      Object {
        "id": 2,
        "label": "snack",
        "typeid": 4,
      },
    ],
    "timing": Object {
      "prep": Array [
        Object {
          "units": "MINUTE",
          "value": "2",
        },
      ],
      "total": Array [
        Object {
          "units": "MINUTE",
          "value": "20",
        },
        Object {
          "units": "HOUR",
          "value": "1",
        },
      ],
    },
    "title": "my first recipe",
  },
  "success": true,
}
`;

exports[`recipeMutationReducer no recipe field 1`] = `
Object {
  "message": "Success!",
  "recipe": null,
  "success": true,
}
`;

exports[`recipeMutationReducer no success field 1`] = `
Object {
  "message": "Success!",
  "recipe": Object {
    "dateAdded": "2020-05-10 00:00:45.511 +00:00",
    "dateUpdated": "2020-05-10 00:01:45.511 +00:00",
    "description": "the best recipe ever",
    "directions": Array [
      Object {
        "label": "section 1",
        "steps": Array [
          Object {
            "text": "step 1",
          },
          Object {
            "text": "step 2",
          },
        ],
      },
    ],
    "id": 123,
    "ingredients": Array [
      Object {
        "ingredients": Array [
          Object {
            "amount": "2",
            "name": "apples",
            "optional": true,
            "prep": "chopped",
            "toTaste": true,
            "unit": "cup",
          },
          Object {
            "amount": Object {
              "max": "10",
              "min": "5",
            },
            "name": "garlic",
            "optional": false,
            "prep": "minced",
            "toTaste": true,
            "unit": "tablespoon",
          },
        ],
        "label": "ingredient section 1",
      },
    ],
    "photo": "http://some.url.com/pic.jpg",
    "servings": 2,
    "source": Object {
      "display": "Mandy's Kitchen",
      "url": "http://some.url.com/recipe",
    },
    "tags": Array [
      Object {
        "id": 1,
        "label": "bread",
        "typeid": 2,
      },
      Object {
        "id": 2,
        "label": "snack",
        "typeid": 4,
      },
    ],
    "timing": Object {
      "prep": Array [
        Object {
          "units": "MINUTE",
          "value": "2",
        },
      ],
      "total": Array [
        Object {
          "units": "MINUTE",
          "value": "20",
        },
        Object {
          "units": "HOUR",
          "value": "1",
        },
      ],
    },
    "title": "my first recipe",
  },
  "success": false,
}
`;

exports[`recipeReducer all recipe components 1`] = `
Object {
  "dateAdded": "2020-05-10 00:00:45.511 +00:00",
  "dateUpdated": "2020-05-10 00:01:45.511 +00:00",
  "description": "the best recipe ever",
  "directions": Array [
    Object {
      "label": "section 1",
      "steps": Array [
        Object {
          "text": "step 1",
        },
        Object {
          "text": "step 2",
        },
      ],
    },
  ],
  "id": 123,
  "ingredients": Array [
    Object {
      "ingredients": Array [
        Object {
          "amount": "2",
          "name": "apples",
          "optional": true,
          "prep": "chopped",
          "toTaste": true,
          "unit": "cup",
        },
        Object {
          "amount": Object {
            "max": "10",
            "min": "5",
          },
          "name": "garlic",
          "optional": false,
          "prep": "minced",
          "toTaste": true,
          "unit": "tablespoon",
        },
      ],
      "label": "ingredient section 1",
    },
  ],
  "photo": "http://some.url.com/pic.jpg",
  "servings": 2,
  "source": Object {
    "display": "Mandy's Kitchen",
    "url": "http://some.url.com/recipe",
  },
  "tags": Array [
    Object {
      "id": 1,
      "label": "bread",
      "typeid": 2,
    },
    Object {
      "id": 2,
      "label": "snack",
      "typeid": 4,
    },
  ],
  "timing": Object {
    "prep": Array [
      Object {
        "units": "MINUTE",
        "value": "2",
      },
    ],
    "total": Array [
      Object {
        "units": "MINUTE",
        "value": "20",
      },
      Object {
        "units": "HOUR",
        "value": "1",
      },
    ],
  },
  "title": "my first recipe",
}
`;

exports[`tagsReducer properly reduces tags 1`] = `
Array [
  Object {
    "id": 1,
    "label": "bread",
    "typeid": 2,
  },
  Object {
    "id": 2,
    "label": "snack",
    "typeid": 4,
  },
]
`;

exports[`timeReducer properly reduces total time array 1`] = `
Array [
  Object {
    "units": "MINUTE",
    "value": "20",
  },
  Object {
    "units": "HOUR",
    "value": "1",
  },
]
`;

exports[`timingReducer properly reduces timing array 1`] = `
Object {
  "prep": Array [
    Object {
      "units": "MINUTE",
      "value": "2",
    },
  ],
  "total": Array [
    Object {
      "units": "MINUTE",
      "value": "20",
    },
    Object {
      "units": "HOUR",
      "value": "1",
    },
  ],
}
`;
